src/containers/UserMenu/UserMenu.container.ts

```diff
@@ -0,0 +1,61 @@
+import { connect } from 'react-redux'
+import { isPending } from '../../modules/transaction/utils'
+import {
+  getAddress,
+  getMana,
+  isConnected,
+  isConnecting,
+  getManaL2
+} from '../../modules/wallet/selectors'
+import {
+  connectWalletRequest,
+  disconnectWallet
+} from '../../modules/wallet/actions'
+import { getData as getProfiles } from '../../modules/profile/selectors'
+import { isEnabled } from '../../modules/translation/selectors'
+import { getTransactions } from '../../modules/transaction/selectors'
+import {
+  MapStateProps,
+  MapDispatch,
+  MapDispatchProps,
+  OwnProps
+} from './UserMenu.types'
+import UserMenu from './UserMenu'
+
+const mapState = (state: any): MapStateProps => {
+  const address = getAddress(state)
+  const profile = getProfiles(state)[address!]
+  return {
+    address,
+    mana: getMana(state),
+    manaL2: getManaL2(state),
+    avatar: profile ? profile.avatars[0] : undefined,
+    isSignedIn: isConnected(state),
+    isSigningIn: isConnecting(state),
+    hasActivity: getTransactions(state, address || '').some(tx =>
+      isPending(tx.status)
+    ),
+    hasTranslations: isEnabled(state)
+  }
+}
+
+const mapDispatch = (dispatch: MapDispatch): MapDispatchProps => ({
+  onSignIn: () => dispatch(disconnectWallet()),
+  onSignOut: () => dispatch(connectWalletRequest())
```

Comment:
I these two are flipped!
Also, should the sign in action be an enable/login modal?

Reply:
Fixed! I also removed the mapping to the onSignIn, since actually many things could happen there, most of our dapps just take you to a `/sign-in` page for example. But it's not very important since we don't do the sign in via this component (we just use it once the user is already signed in). The actual sign in button that users click is the one thats part of the Navbar component